base/compiler/abstractinterpretation.jl

```diff
@@ -592,6 +599,10 @@ function abstract_iteration(interp::AbstractInterpreter, @nospecialize(itft), @n
             break
         end
         if nounion.parameters[1] <: valtype && nounion.parameters[2] <: statetype
+            if typeintersect(stateordonet, Nothing) === Union{}
+                # Reached a fixpoint, but Nothing is not possible => iterator is infinite or failing
+                return Any[Bottom]
```

Comment:
In the first iteration of this loop, the `stateordonet` has been computed using possibly const arguments, so while it has been `widenconst`ed, the type may be misleading here. E.g. just `(1:33...,)` will be infered as `Union{}` due to this.